openedx_learning/apps/authoring/containers/api.py

```diff
@@ -99,41 +108,35 @@ def create_next_defined_list(
         # 2. Only associate existent rows to the new entity list iff: the order is the same, the PublishableEntity is in entity_pks and versions are not pinned
         # 3. If the order is different for a row with the PublishableEntity, create new row with the same PublishableEntity for the new order
         # and associate the new row to the new entity list
-        current_rows = previous_entity_list.entitylistrow_set.all()
-        publishable_entities_in_rows = {row.entity.pk: row for row in current_rows}
         new_rows = []
-        for order_num, entity_pk, draft_version_pk, published_version_pk in zip(
-            order_nums, entity_pks, draft_version_pks, published_version_pks
+        for order_num, entity_pk, entity_version_pk in zip(
+            order_nums, entity_pks, entity_version_pks
         ):
-            row = publishable_entities_in_rows.get(entity_pk)
-            if row and row.order_num == order_num:
-                new_entity_list.entitylistrow_set.add(row)
-                continue
```

Comment:
This had a bug - the code was _moving_ `EntityListRow` objects into the new `EntityList` rather than copying them into it.